src/collapse/xdebug.rs

```diff
@@ -0,0 +1,367 @@
+            RawEntryMut::Occupied(occ) => return Interned::Old(Str(*occ.get())),
+            RawEntryMut::Vacant(vacant) => vacant,
+        };
+
+        let index = self.interned_string.len();
+        let element: Rc<str> = Rc::from(string);
+        self.interned_string.push(element.clone());
+        vacant.insert_hashed_nocheck(hash, element, index);
+        Interned::New(Str(index))
+    }
+
+    /// Intern a `Call` into a `usize` index.
+    ///
+    /// Some tests revealed that this was faster than comparing the `Call` instances directly, so
+    /// it was kept. This seems a bit odd but was backed by data at the time.
+    fn intern(&mut self, call: Interned<Call>) -> usize {
+        let new = match call {
+            // The strings were not seen before, definitely new.
+            Interned::New(t) => t,
+            // The strings used were seen before, but maybe not in this call. So retest.
+            Interned::Old(t) => {
+                if let Some(idx) = self.calls.get(&t) {
+                    return *idx;
+                } else {
+                    t
+                }
+            }
+        };
+
+        let index = self.interned.len();
+        self.interned.push(new);
+        self.calls.insert(new, index);
+        index
+    }
+
+    /// Prepare the stack frame for printing.
+    fn frames<'a>(&'a self, stack: &'a [usize]) -> Frames<'a> {
+        Frames { calls: self, stack }
+    }
+
+    /// Create a name for the current stack.
+    ///
+    /// This is potentially costly.
+    fn write_name(&self, indices: &[usize], buffer: &mut fmt::Formatter) -> fmt::Result {
+        let mut indices = indices.iter().cloned();
+        if let Some(first) = indices.by_ref().next() {
+            self.write_call(self.interned[first], buffer)?;
+        }
+        for next in indices {
+            buffer.write_str(";")?;
+            self.write_call(self.interned[next], buffer)?;
+        }
+        Ok(())
+    }
+
+    /// Format a single call.
+    fn write_call(&self, call: Call, buffer: &mut fmt::Formatter) -> fmt::Result {
+        match call {
+            Call::WithoutPath(Str(idx)) => buffer.write_str(&self.interned_string[idx]),
+            Call::WithPath(Str(name), Str(path)) => {
+                let (name, path) = (&self.interned_string[name], &self.interned_string[path]);
+                buffer.write_str(name)?;
+                buffer.write_str("(")?;
+                buffer.write_str(path)?;
+                buffer.write_str(")")
+            }
+        }
+    }
+}
+
+impl<'a> fmt::Display for Frames<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.calls.write_name(self.stack, f)
+    }
+}
+
+impl<'a> Fields<'a> {
+    pub fn new(line: &'a str) -> Self {
+        Fields { line }
+    }
+}
+
+impl<'a> Iterator for Fields<'a> {
+    type Item = &'a str;
+
+    fn next(&mut self) -> Option<&'a str> {
+        let begin = self
+            .line
+            .as_bytes()
```

Comment:
Any particular reason you don't use `self.line.as_bytes().split(b'\t')`? That should also be fast since it also basically assumes ASCII.